esp/services/ws_logaccess/WsLogAccessService.cpp

```diff
@@ -240,6 +240,50 @@ bool Cws_logaccessEx::onGetLogs(IEspContext &context, IEspGetLogsRequest &req, I
     logFetchOptions.setLimit(limit);
     logFetchOptions.setStartFrom((offset_t)startFrom);
 
+    if (version >= 1.03)
+    {
+        IArrayOf<IConstSortCondition> & sortby = req.getSortBy();
+        for (int index = 0; index < sortby.length(); index++)
+        {
+            IConstSortCondition& condition = sortby.item(index);
+            CSortDirection espdirection = condition.getDirection();
+            SortByDirection direction = SORTBY_DIRECTION_none;
+            if (espdirection == CSortDirection_ASC)
```

Comment:
Agreed. I would have used a common enum if supported. I suspect it is not supported, or not worth doing due to other dependency complications. Will reach out to @afishbeck .

Reply:
I don't think it would be easy to automatically sync with a c++ enum since we would need to be able to enumerate the names and values in ESDL somehow.  We could standardize on what would need to be in jlib to make this do able.  For example a parallel names array or some such. 

But,

We could probably make improvements to the ESDL enum stuff to allow you to:
1. Set the enum value using the jlib enum value (this might already be possible?).
2. Then if the enum value was set that way, use the actual enum value names in generated code.
3. If it's a named enum, class, or type define, allow you to specify it as the return type of the "get" function.